nmdc_server/jobs.py

```diff
@@ -20,25 +19,34 @@ def ping():
     return True
 
 
-def migrate(database_uri):
+def migrate(ingest_db: bool = False):
     """Update the database to the latest HEAD.
 
     This function will also create the schema if necessary.
     """
-    engine = create_engine(database_uri)
-    with create_session(engine=engine) as db:
+    if ingest_db:
+        database_uri = settings.ingest_database_uri
+        engine = database.engine_ingest
+        session_maker = database.SessionLocalIngest
+    else:
+        database_uri = settings.current_db_uri
+        engine = database.engine
+        session_maker = database.SessionLocal
+
+    with session_maker() as db:
         alembic_cfg = Config(str(HERE / "alembic.ini"))
         alembic_cfg.set_main_option("script_location", str(HERE / "migrations"))
         alembic_cfg.set_main_option("sqlalchemy.url", database_uri)
         alembic_cfg.attributes["configure_logger"] = True
 
-        context = MigrationContext.configure(db.connection())
-        head = context.get_current_revision()
-        if head is None:
-            metadata.create_all(engine)
-            command.stamp(alembic_cfg, "head")
-        else:
-            command.upgrade(alembic_cfg, "head")
+        with db.connection() as conn:
```

Comment:
This change has a reasonable probability of fixing the connection leaking issue. `db.connection()` is a context manager that is supposed to be closed at the end.